src/__tests__/components/molecules/Redirect.test.tsx

```diff
@@ -0,0 +1,40 @@
+import { mockDeep } from 'jest-mock-extended';
+import { render } from '@testing-library/react';
+import { Redirect } from '@molecules/Redirect/Redirect';
+import {
+  useSearchParams,
+  ReadonlyURLSearchParams,
+  redirect,
+} from 'next/navigation';
+
+jest.mock('next/navigation', () => ({
+  ...jest.requireActual('next/navigation'),
+  redirect: jest.fn(),
+  useSearchParams: jest.fn(),
+}));
+
+test('Redirect - URL provided', () => {
+  const mockRedirect = jest.fn(mockDeep<typeof redirect>());
```

Comment:
You shouldn't need to mock deep this function.

Doing;

```
const redirectMock = jest.mocked(redirect)
```

Should be fine. I've used the same pattern in other tests.

Reply:
updated